### PR TITLE
fix the concurrency issue of Logger and minor fix the test

### PR DESCRIPTION
**Pull request by ansyral**
1. Logger.RegisterListener and UnRegisterListener has concurrency issue.
2. add phase scope for the test to avoid data conflict since unit tests in different test classes run in parallel.
@vwxyzh @superyyrrzz @hellosnow @qinezh @chenkennt

### CODE REVIEW

File: test/Microsoft.DocAsCode.Common.Tests/ScopeTest.cs [hunk 1]
```diff
@@ -87,61 +87,61 @@ public void TestPhaseScope()
...
         {
-            var listener = new TestLoggerListener();
+            var listener = new TestLoggerListener("TestFileScope");
             try
             {
                 Logger.RegisterListener(listener);
                 Action callback;
 
-                Logger.LogInfo("Not in file scope.");
+                Logger.LogInfo("Not in file scope.", phase: "TestFileScope");
```
Comment (hellosnow) [inline test/Microsoft.DocAsCode.Common.Tests/ScopeTest.cs]: >"TestFileScope" [](start = 60, length = 15)

listener.Phase

**Comment by hellosnow** [comment]
:shipit:

File: test/Microsoft.DocAsCode.Common.Tests/ScopeTest.cs (hunk 1 above)
Comment (ansyral) [inline test/Microsoft.DocAsCode.Common.Tests/ScopeTest.cs]: Resolved

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**